packages/relay-runtime/store/RelayReader.js

```diff
@@ -1111,11 +1113,40 @@ class RelayReader {
     const inlineData = {};
     const parentFragmentName = this._fragmentName;
     this._fragmentName = fragmentSpreadOrFragment.name;
+
+    const parentVariables = this._variables;
+
+    // If the inline fragment spread has arguments, we need to temporarily
+    // switch this._variables to include the fragment spread's arguments
+    // for the duration of its traversal.
+    if (fragmentSpreadOrFragment.args) {
+      const argumentVariables = getArgumentValues(
+        fragmentSpreadOrFragment.args,
+        this._variables,
+      );
+
+      this._variables = RelayConcreteVariables.getFragmentVariables(
+        fragmentSpreadOrFragment,
+        this._owner.variables,
+        argumentVariables,
+      );
+    } else {
+      this._variables = RelayConcreteVariables.getFragmentVariables(
+        fragmentSpreadOrFragment,
+        this._owner.variables,
+        {},
+      );
+    }
```

Comment:
Style nit: Would this work?
```suggestion
    const argumentVariables = fragmentSpreadOrFragment.args ? getArgumentValues(
      fragmentSpreadOrFragment.args,
      this._variables,
    ) : {};

    this._variables = RelayConcreteVariables.getFragmentVariables(
      fragmentSpreadOrFragment,
      this._owner.variables,
      argumentVariables,
    );
```